frappe/integrations/doctype/social_login_key/test_social_login_key.py

```diff
@@ -14,6 +20,41 @@ def test_adding_frappe_social_login_provider(self):
 		social_login_key.get_social_login_provider(provider_name, initialize=True)
 		self.assertRaises(BaseUrlNotSetError, social_login_key.insert)
 
+	def test_github_login_with_private_email(self):
+		github_social_login_setup()
+
+		mock_session = MagicMock()
+		mock_session.get.side_effect = github_response_for_private_email
+
+		with patch.object(OAuth2Service, "get_auth_session", return_value=mock_session):
+			login_via_oauth2("github", "iwriu", {"token": "ewrwerwer"})			# Dummy code and state token
```

Comment:
better to have some assertion... 
you can assert if the expected response after successful login was set in `frappe.local.response`

Reply:
do not want to hold this PR just for this... merging this for now since it is a critical fix... 
I would suggest you take care of it next time.